### PR TITLE
Implement WindowManager::updateDaily and ObjectManager::reloadAll

### PR DESCRIPTION
**Pull request by duncanspumpkin**
A selection of small functions in the end.

### CODE REVIEW

File: src/OpenLoco/src/Objects/ObjectManager.cpp [hunk 1]
```diff
@@ -559,6 +536,87 @@ namespace OpenLoco::ObjectManager
...
     }
 
+    // 0x00471B95
+    void freeTemporaryObject()
+    {
+        if (_temporaryObject != nullptr && _temporaryObject != reinterpret_cast<Object*>(-1))
+        {
+            free(_temporaryObject);
+            _temporaryObject = reinterpret_cast<Object*>(-1);
+        }
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Objects/ObjectManager.cpp:546]: This is keeping -1 for vanilla compatibility, I presume? Perhaps add a note to that end?

File: src/OpenLoco/src/Objects/ObjectManager.cpp [hunk 2]
```diff
@@ -559,6 +536,87 @@ namespace OpenLoco::ObjectManager
...
+        const auto filePath = Environment::getPath(Environment::PathId::objects) / fs::u8path(installedObject->_filename);
+
+        SawyerStreamReader stream(filePath);
+        ObjectHeader loadingHeader;
+        stream.read(&loadingHeader, sizeof(loadingHeader));
+        if (loadingHeader != header)
+        {
+            // Something wrong has happened and installed object does not match index
+            // Vanilla continued to search for subsequent matching installed headers.
+            return false;
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Objects/ObjectManager.cpp:568]: Perhaps print to console in such cases?

File: src/OpenLoco/src/Objects/ObjectManager.cpp [hunk 3]
```diff
@@ -559,6 +536,87 @@ namespace OpenLoco::ObjectManager
...
+        }
+
+        // Vanilla would branch and perform more efficient readChunk if size was known from installedObject.ObjectHeader2
+        auto data = stream.readChunk();
+
+        if (!computeObjectChecksum(loadingHeader, data))
+        {
+            // Something wrong has happened and installed object checksum is broken
+            return false;
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Objects/ObjectManager.cpp:577]: Perhaps print to console in such cases?

File: src/OpenLoco/src/Objects/ObjectManager.cpp (hunk 1 above)
Comment (duncanspumpkin) [reply src/OpenLoco/src/Objects/ObjectManager.cpp]: yeah unfortunately still needed.

Comment (ZehMatt) [reply src/OpenLoco/src/Objects/ObjectManager.cpp]: Could have
```cpp
static const auto kInvalidObject = reinterpret_cast<Object*>(-1);
```
instead.

**Review by AaronVanGeffen — CHANGES_REQUESTED** [review]
The build for PR crashes for me, directly after building the object index:
```
Create progress bar
SendMessage(PBM_SETRANGE, 256)
SendMessage(PBM_SETSTEP, 1)
SendMessage(PBM_SETPOS, 60)
SendMessage(PBM_SETPOS, 220)
SendMessage(PBM_SETPOS, 235)
SendMessage(PBM_SETPOS, 250)
Destroy progress bar
malloc(): unaligned tcache chunk detected
Process 3636 stopped
* thread #1, name = 'OpenLoco', stop reason = signal SIGABRT
    frame #0: 0xf7fc6549 [vdso]`__kernel_vsyscall + 9
[vdso]`__kernel_vsyscall:
->  0xf7fc6549 <+9>:  popl   %ebp
    0xf7fc654a <+10>: popl   %edx
    0xf7fc654b <+11>: popl   %ecx
    0xf7fc654c <+12>: retl   
(lldb) bt
* thread #1, name = 'OpenLoco', stop reason = signal SIGABRT
  * frame #0: 0xf7fc6549 [vdso]`__kernel_vsyscall + 9
    frame #1: 0xf7726667 libc.so.6`___lldb_unnamed_symbol3803 + 279
    frame #2: 0xf76d2677 libc.so.6`raise + 39
    frame #3: 0xf76bb126 libc.so.6`abort + 242
    frame #4: 0xf77186a4 libc.so.6`___lldb_unnamed_symbol3698 + 628
    frame #5: 0xf7731d2f libc.so.6`___lldb_unnamed_symbol3871 + 31
    frame #6: 0xf7736403 libc.so.6`__libc_malloc + 691
    frame #7: 0x011f6bf4 OpenLoco`::fn_malloc(size=0) at Hooks.cpp:330:18
(lldb) 
```

The object index gets built, however there are a few minor differences. Not sure if these can be attributed to clearing out padding?
Screenshot_20221116_210753

**Review by AaronVanGeffen — APPROVED** [review]
Looks like this is working fine.

**Pull request merged**